src/lyra/semantics/semantics.py

```diff
@@ -303,6 +378,62 @@ def range_call_semantics(self, stmt: Call, state: State) -> State:
         error = f"Call to {stmt.name} with unexpected number of arguments!"
         raise ValueError(error)
 
+    def items_call_semantics(self, stmt: Call, state: State) -> State:
+        """Semantics of calls to 'items'.
+
+                :param stmt: call to 'items' to be executed
+                :param state: state before executing the call statement
+                :return: state modified by the call statement
+                """
+        if isinstance(stmt.arguments[0], VariableAccess):  # target
+            state.result = {Items(stmt.typ, stmt.arguments[0].variable)}
+        else:
+            error = f"Semantics for items() call on non-variable {stmt.arguments[0]} is not yet " \
+                    f"implemented!"
+            raise NotImplementedError(error)
+        return state
+
+    def keys_call_semantics(self, stmt: Call, state: State) -> State:
+        """Semantics of calls to 'keys'.
+
+                        :param stmt: call to 'keys' to be executed
+                        :param state: state before executing the call statement
+                        :return: state modified by the call statement
+                        """
+        if isinstance(stmt.arguments[0], VariableAccess):  # target
+            state.result = {Keys(stmt.typ, stmt.arguments[0].variable)}
+        else:
+            error = f"Semantics for keys() call on non-variable {stmt.arguments[0]} is not yet " \
+                    f"implemented!"
+            raise NotImplementedError(error)
+        return state
+
+    def values_call_semantics(self, stmt: Call, state: State) -> State:
+        """Semantics of calls to 'values'.
+
+                        :param stmt: call to 'values' to be executed
+                        :param state: state before executing the call statement
+                        :return: state modified by the call statement
+                        """
+        if isinstance(stmt.arguments[0], VariableAccess):     # target
+            state.result = {Values(stmt.typ, stmt.arguments[0].variable)}
+        else:
+            error = f"Semantics for values() call on non-variable {stmt.arguments[0]} is not yet "\
+                    f"implemented!"
+            raise NotImplementedError(error)
+        return state
+
+    # TODO: define default call semantics instead for usage?
+    def lower_call_semantics(self, stmt: Call, state: State) -> State:
```

Comment:
Why is it called `lower_call_semantcs`?

Reply:
I added it for the example from my description, where there is a call to lower() on a string, which just puts everything in lower case letters.

Reply:
Ah ok, so it is just the docstring that is outdated then.

Reply:
Yes :-)